fe/src/views/world-header.view.tsx

```diff
@@ -0,0 +1,126 @@
+import { Header, If } from "../utils/garage";
+import { LanguageToggleView } from "./language-toggle.view";
+import { route } from "preact-router";
+import { Breadcrumb, Tab, Tabs } from "../utils/fe-utils";
+import { PageProps } from "../utils/page-wrapper";
+import { useState } from "preact/hooks";
+import ReturnButtonView from "./return-button.view";
+
+const selectClass = (input: string): { class: string; selected: boolean } => {
+  const baseClass = `rounded-xl p-1rem`;
+  if (window.location.pathname.includes(input)) {
+    return { class: `bg-#0046FF font-normal ${baseClass}`, selected: true };
+  } else {
+    return { class: baseClass, selected: false };
+  }
+};
+
+const getRoute = (input: string) => {
+  if (input === "world") return "world";
+  if (input === "people") return "article?people=1";
+  if (input === "selves") return "selves";
+  return input;
+};
+
+export function WorldHeaderView({ routerInfo }: PageProps) {
+  const [inputResearchValue, setInputResearchValue] = useState<string>("");
+  const languageLabel = routerInfo.lang?.toUpperCase();
+  const Options = (props: { en: string; zh: string }) => {
+    const { en, zh } = props;
+    return (
+      <Tab
+        value={en}
+        class={selectClass(en).class}
+        selected={selectClass(en).selected}
+      >
+        <If condition={languageLabel}>
+          <div slot="EN">{en.toUpperCase()}</div>
+          <div slot="ZH">{zh}</div>
+        </If>
+      </Tab>
+    );
+  };
+
+  const handleInputChange = (e: Event) => {
+    const target = e.target as HTMLInputElement;
+    setInputResearchValue(target.value);
+  };
+
+  const handleEnterPress = (e: KeyboardEvent) => {
+    if (e.key === "Enter") {
+      route(
+        `/${routerInfo.lang}/article?search=${encodeURIComponent(
+          inputResearchValue
+        )}`
+      );
+    }
+  };
+
+  const checkHeaderInputVisibility = () => {
+    if (routerInfo.path === "/:lang/world") {
```

Comment:
this can be convert to 
```
if (condition 1) return true
if (condition 2) return true
return false
```
for better readability

Reply:
right. learn it!

Reply:
done

Reply:
Actually, I wonder why there is a case for ":lang/article" in the World page. In my opinion, even if we need to ckeck a condition in this page, it should be either ":lang/world" or ":lang/world/:area".

Reply:
牛逼。。。。 学会了。这咋一个比一个牛逼呢

Reply:
the General Article List also use world header

> Actually, I wonder why there is a case for ":lang/article" in the World page. In my opinion, even if we need to ckeck a condition in this page, it should be either ":lang/world" or ":lang/world/:area". Point out if I was wrong.
> 
> By the way, in this specific case, I would personally prefer if-without-else over switch-case because it is actually just two cases. But switch-case also looks good to me.
> 
> ```ts
> const whateverCases = ["something", "something else"] ;
> if(whateverCases.includes(parameter)) return true;
> return false;
> ```
> 
> For lambda functions, you may even just do below since itself returns a boolean value.`includes()`
> 
> ```ts
> (parameter) => ["something", "something else"].includes(parameter);
> ```

Reply:
If I were you, I would rether copy/paste it to make a duplicate than add a case here with confusing  component naming. 🙄

Reply:
done

Reply:
我们最终决定使用if return的形式来解决判断，这样更容易阅读和理解。